association_backend/association/domain/entities/subscription_entities.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+import pydantic
+from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Table
+from sqlalchemy.orm import registry, relationship
+
+
+class UserData(pydantic.BaseModel):
+    email: str
+    user_id: int
+
+
+class SubscriptionState(str, Enum):
+    PENDING = "pending"
+    ACTIVE = "active"
+    EXPIRED = "expired"
+    NOT_CREATED = "not-created"
```

Comment:
what does `not-created` mean?

Reply:
After a certain amount of time, the checkoutSession is no longer valuable, so the system needs to know this in order to create a new one.
We have 2 solutions: use the checkoutSession as long as it is valid and manage the invalidation through a "subscription_update" callback (what is implemented now), or create a checkoutSession at each user click by deleting the old one (I have not analyzed well how we could manage this other flow)
Using the first solution, when the session expires, not having been paid on time, the system will delete the stripe_session_id from our Subscription in order to create a new one the next time.
Probably "NOT_CREATED" is not the right term, perhaps "FIRST_PAYMENT_EXPIRED" would be better
Do you think this name Is better?
We may also need it for monitoring purposes

Reply:
I renamed it in `FIRST_PAYMENT_EXPIRED`